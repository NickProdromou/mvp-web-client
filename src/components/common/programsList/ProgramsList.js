import React, { Component } from 'react';
import { Page, Row, Column } from 'hedron/lib/index';
import PropTypes from 'prop-types';
import ProgramCard from '../programCard/programCard'
import ProgramListHeader from './programsList.style';

export default class ProgramsList extends Component {

    static propTypes = {
      programs: PropTypes.arrayOf(PropTypes.shape({
          title: PropTypes.string,
          keys: PropTypes.number
      }))
    };

    static contextTypes = {
        router: PropTypes.func.isRequired
    };

    static defaultProps = {
      programs:  [
          {title:"photoshop", keys: 334,link: {to: '/programs/photoshop', text:'view'}},
          {title:"Illustrator", keys: 334,link: {to: '/programs/illustrator', text:'view'}},
          {title:"Chrome", keys: 334,link: {to: '/programs/chrome', text:'view'}},
          {title:"Webstorm", keys: 334,link: {to: '/programs/webstorm', text:'view'}},
          {title:"OSX", keys: 334,link: {to: '/programs/osx', text:'view'}},
          {title:"Iterm2", keys: 334,link: {to: '/programs/iterm', text:'view'}},
      ]
    };

    render() {

        const { programs } = this.props;

        return (
        <div>
            <ProgramListHeader fluid>
                <Row>
                    <Column>
                        <h2>Programs</h2>
                        <p>Pick a program, Any Program</p>
                    </Column>
                </Row>
            </ProgramListHeader>
            <Page>
                <Row>
                    { programs.map( (program, index)  => (
                        <ProgramCard
                            index={index}
                            title={program.title}
                            keys={program.keys}
                            link={program.link}
                        />
                    )) }
                </Row>
            </Page>
        </div>

        )
    }

}

