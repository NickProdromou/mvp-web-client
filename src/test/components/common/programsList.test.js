import React from 'react';
import ProgramsList from '../../../components/common/programsList/ProgramsList';
import renderer from 'react-test-renderer';

it('renders correctly', () => {

    const tree = renderer.create(
        <ProgramsList/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
