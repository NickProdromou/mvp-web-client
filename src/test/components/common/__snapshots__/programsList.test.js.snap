// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="sc-bZQynM kTlMAn sc-bxivhb jcNPuI"
  >
    <section
      className="sc-bwzfXH irrVft"
    >
      <div
        className="sc-htpNat bKOCBb"
      >
        <h2>
          Programs
        </h2>
        <p>
          Pick a program, Any Program
        </p>
      </div>
    </section>
  </div>
  <div
    className="sc-bxivhb bIAnAi"
  >
    <section
      className="sc-bwzfXH irrVft"
    >
      <div
        className="sc-EHOje LlIEo sc-htpNat egfJaR"
      >
        <article>
          <h3>
            photoshop
          </h3>
          <p>
            Keyboard shortcuts: 
            334
             
          </p>
          <a
            className="sc-ifAKCX kZILIl"
            href="/sample-link"
          >
            view
          </a>
        </article>
      </div>
      <div
        className="sc-EHOje LlIEo sc-htpNat egfJaR"
      >
        <article>
          <h3>
            Illustrator
          </h3>
          <p>
            Keyboard shortcuts: 
            334
             
          </p>
          <a
            className="sc-ifAKCX kZILIl"
            href="/sample-link"
          >
            view
          </a>
        </article>
      </div>
      <div
        className="sc-EHOje LlIEo sc-htpNat egfJaR"
      >
        <article>
          <h3>
            Chrome
          </h3>
          <p>
            Keyboard shortcuts: 
            334
             
          </p>
          <a
            className="sc-ifAKCX kZILIl"
            href="/sample-link"
          >
            view
          </a>
        </article>
      </div>
      <div
        className="sc-EHOje LlIEo sc-htpNat egfJaR"
      >
        <article>
          <h3>
            Webstorm
          </h3>
          <p>
            Keyboard shortcuts: 
            334
             
          </p>
          <a
            className="sc-ifAKCX kZILIl"
            href="/sample-link"
          >
            view
          </a>
        </article>
      </div>
      <div
        className="sc-EHOje LlIEo sc-htpNat egfJaR"
      >
        <article>
          <h3>
            OSX
          </h3>
          <p>
            Keyboard shortcuts: 
            334
             
          </p>
          <a
            className="sc-ifAKCX kZILIl"
            href="/sample-link"
          >
            view
          </a>
        </article>
      </div>
      <div
        className="sc-EHOje LlIEo sc-htpNat egfJaR"
      >
        <article>
          <h3>
            Iterm2
          </h3>
          <p>
            Keyboard shortcuts: 
            334
             
          </p>
          <a
            className="sc-ifAKCX kZILIl"
            href="/sample-link"
          >
            view
          </a>
        </article>
      </div>
    </section>
  </div>
</div>
`;
